runtime/src/tasks/fifo/tasks-fifo.c

```diff
@@ -1497,3 +1535,105 @@ uint32_t chpl_task_getNumIdleThreads(void) {
   assert(numIdleThreads >= 0);
   return numIdleThreads;
 }
+
+
+//
+// Tasking callback support.
+//
+int chpl_task_install_callback(chpl_task_cb_event_kind_t event_kind,
+                               chpl_task_cb_info_kind_t info_kind,
+                               chpl_task_cb_fn_t cb_fn) {
+  int i;
+
+  if (event_kind >= chpl_task_cb_num_event_kinds) {
+    errno = ERANGE;
+    return -1;
+  }
+
+  i = cb_info[event_kind].count;
+
+  if (i >= MAX_CBs_PER_EVENT) {
+    errno = ENOMEM;
+    return -1;
+  }
+
+  cb_info[event_kind].count++;
+  cb_info[event_kind].fns[i]= cb_fn;
+  cb_info[event_kind].info_kinds[i] = info_kind;
+
+  return 0;
+}
+
+
+int chpl_task_uninstall_callback(chpl_task_cb_event_kind_t event_kind,
+                                 chpl_task_cb_fn_t cb_fn) {
+  int i;
+
+  if (event_kind >= chpl_task_cb_num_event_kinds) {
+    errno = ERANGE;
+    return -1;
+  }
+
+  for (i = 0; i < cb_info[event_kind].count; i++) {
```

Comment:
It took me a little to get what was going on here. Personally I would prefer something like:

```c
for (i = 0; i < cb_info[event_kind].count; i++) {
  if (cb_info[event_kind].fns[i] == cb_fn) {
    event_index = i;
    break;
  }
}

// use event_index instead of re-using i
```